core/inputs-texlike.lua

```diff
@@ -7,8 +7,9 @@ local ID = lpeg.C(SILE.parserBits.letter * (SILE.parserBits.letter + SILE.parser
 SILE.inputs.TeXlike.identifier = (ID + lpeg.S":-")^1
 
 SILE.inputs.TeXlike.passthroughCommands = {
+  ftl = true,
```

Comment:
Since these keywords are defined at parsing level, does it mean they are kind of reserved and we cannot afterwards define commands by that name? It does seem its the case (e.g. we cannot seem to `define` a command named math or script and expect it to work correctly... If this is right, it sounds like code smell (if each time we want to define pass-through commands, we make them impossible to exist in input files and be redefined). We are eventually gonna cram the core parser with plenty of path-through commands, no? (if we start adding pass-through this way for tikz, svg, lilypond, etc.)...

Reply:
Yes, it does make those commands reserved.

The issue here is that the parser digests a full document into an AST tree before it starts walking through it running commands. Nothing inside the commands (e.g. you redefining `math` in a macro or Lua code can change how the AST was built because it is already built before any processing gets done.

You don't just need to redefine the `math`/`script`/`ftl` commands than run when such a node is hit in the AST, you need to redefine the grammar of the language that parses and builds the AST.

If you have a way to solve that cart/horse problem I'm all ears.

Reply:
Indeed. I have commented there on the general parsing issue: https://github.com/sile-typesetter/sile/issues/1092#issuecomment-1065583740 - very rough idea out of my (no so) fresh mind, scrap it if I got it wrong !

Reply:
I'm marking this thread resolved only for the purposes of *this PR*, the topic we got on here is very much still in discussion and I dropped [some clips and links in the relevant issue](https://github.com/sile-typesetter/sile/issues/1092#issuecomment-1065826926) so they don't get lost.